packages/otp2-tile-overlay/src/index.tsx

```diff
@@ -0,0 +1,178 @@
+import type { EventData } from "mapbox-gl"
+import { LAYER_PAINT } from "./util"
+
+const OTP2TileLayerWithPopup = ({
+  configCompanies,
+  id,
+  network,
+  setLocation,
+  setViewedStop,
+  type
+}: {
+  /**
+   * Optional configuration item which allows for customizing properties of scooter and
+   * bikeshare companies
+   */
+  configCompanies?: ConfiguredCompany[]
+  id: string
+  name?: string
+  /**
+   * If `network` is specified, the layer will be filtered to only show vehicles from
+   * that network
+   */
+  network?: string
+  /**
+   * A method fired when a stop is selected as from or to in the popup
+   */
+  setLocation: (location: MapLocationActionArg) => void
+  /**
+   * A method fired when the stop viewer is opened in the popup
+   */
+  setViewedStop: ({ stopId }: { stopId: string }) => void
+  type: string
+}): JSX.Element => {
+  const { current: map } = useMap()
+
+  // TODO: handle this complex type: it can be a stop, a station, and some extra fields too
+  const [clickedEntity, setClickedEntity] = useState<any>(null)
+  useEffect(() => {
+    map?.on("mouseenter", id, () => {
+      map.getCanvas().style.cursor = "pointer"
+    })
+    map?.on("mouseleave", id, () => {
+      map.getCanvas().style.cursor = ""
+    })
+    map?.on("click", id, (event: EventData) => {
+      const { sourceLayer } = event.features?.[0]
+      const synthesizedEntity = {
+        ...event.features?.[0].properties,
+        lat: event.lngLat.lat,
+        lon: event.lngLat.lng,
+        sourceLayer
+      }
+
+      if (sourceLayer === "stops" || sourceLayer === "stations") {
+        setClickedEntity(synthesizedEntity)
+      }
+      if (
+        sourceLayer === "rentalVehicles" ||
+        sourceLayer === "rentalStations"
+      ) {
+        setClickedEntity({
+          // GraphQL field not in the tile info, but we can deduce it
+          isFloatingBike:
+            sourceLayer === "rentalVehicles" &&
+            synthesizedEntity.formFactor === "BICYCLE",
+          // GraphQL field not in the tile info, but we can deduce it
+          isFloatingVehicle:
+            sourceLayer === "rentalVehicles" &&
+            synthesizedEntity.formFactor === "SCOOTER",
+          // OTP1 compatibility -- will get overwritten if possible
+          networks: [synthesizedEntity.network],
+          ...synthesizedEntity,
+          // OTP1 compatibility
+          bikesAvailable: synthesizedEntity.vehiclesAvailable,
+          // OTP1 compatibility
+          x: synthesizedEntity.lon,
+          // OTP1 compatibility
+          y: synthesizedEntity.lat
+        })
+      }
+    })
+  }, [id, map])
+
+  return (
+    <>
+      <Layer
+        filter={network ? ["all", ["==", "network", network]] : ["all"]}
+        id={id}
+        key={id}
+        paint={LAYER_PAINT[type]}
+        source="otp2-tiles"
+        source-layer={type}
+        type="circle"
+      />
+      {clickedEntity && (
+        <Popup
```

Comment:
I feel strongly about having this package not display popups at all. Actually, we had a related team discussion about removing popups from the `endpoint-overlay` package that goes in the same direction. 

The reason for that was to not pollute the component with handlers and contents that are implementer-specific, such as `setLocation` or the popups below.

We should instead publish an event such as `onEntityClick`, and demo handling these events in storybook using popups.

Reply:
I've made the new props optional so that the layer can operate in a "minimal" mode where only the title is displayed.